File: vocabulary game/Social Media Game/inject.py
import json
import os

vocab_files = ["nouns.json", "nouns2.json", "verbs.json", "adjectives.json", "others.json"]
all_vocab = []

for file in vocab_files:
    if os.path.exists(file):
        with open(file, 'r', encoding='utf-8') as f:
            all_vocab.extend(json.load(f))

with open("phrases.json", 'r', encoding='utf-8') as f:
    phrases = json.load(f)

vocab_json = json.dumps(all_vocab, ensure_ascii=False)
phrases_json = json.dumps(phrases, ensure_ascii=False)

with open("Social_Media_Game.html", 'r', encoding='utf-8') as f:
    html = f.read()

idx1 = html.find("const allVocab = [")
idx2 = html.find("const phrases = [", idx1)
idx3 = html.find("let particles =", idx2)

if idx1 != -1 and idx2 != -1 and idx3 != -1:
    part1 = html[:idx1]
    part2 = f"const allVocab = {vocab_json};\n\nconst phrases = {phrases_json};\n\n"
    part3 = html[idx3:]
    
    with open("Social_Media_Game.html", 'w', encoding='utf-8') as f:
        f.write(part1 + part2 + part3)
    print("Injected successfully!")
else:
    print("Failed to find indices.")

File: vocabulary game/Social Media Game/inject.js
const fs = require('fs');

const vocabFiles = ['nouns.json', 'nouns2.json', 'verbs.json', 'adjectives.json', 'others.json'];
let allVocab = [];

for (const file of vocabFiles) {
    if (fs.existsSync(file)) {
        const data = JSON.parse(fs.readFileSync(file, 'utf8'));
        allVocab = allVocab.concat(data);
    }
}

const phrases = JSON.parse(fs.readFileSync('phrases.json', 'utf8'));

let html = fs.readFileSync('Social_Media_Game.html', 'utf8');

// Replace allVocab array
const vocabStart = html.indexOf('const allVocab = [');
const vocabEnd = html.indexOf('];\r\n\r\nconst phrases = [', vocabStart);
if(vocabStart > -1 && vocabEnd > -1) {
    html = html.substring(0, vocabStart) + 'const allVocab = ' + JSON.stringify(allVocab, null, 2) + html.substring(vocabEnd + 1);
} else {
    // Try \n instead of \r\n
    const vocabEnd2 = html.indexOf('];\n\nconst phrases = [', vocabStart);
    if(vocabStart > -1 && vocabEnd2 > -1) {
        html = html.substring(0, vocabStart) + 'const allVocab = ' + JSON.stringify(allVocab, null, 2) + html.substring(vocabEnd2 + 1);
    } else {
        // Just find next ];
        const vEnd = html.indexOf('];', vocabStart);
        html = html.substring(0, vocabStart) + 'const allVocab = ' + JSON.stringify(allVocab, null, 2) + html.substring(vEnd + 1);
    }
}

// Replace phrases array
const phraseStart = html.indexOf('const phrases = [');
const phraseEnd = html.indexOf('];', phraseStart);
if(phraseStart > -1 && phraseEnd > -1) {
    html = html.substring(0, phraseStart) + 'const phrases = ' + JSON.stringify(phrases, null, 2) + html.substring(phraseEnd + 1);
}

fs.writeFileSync('Social_Media_Game.html', html, 'utf8');
console.log('Injected successfully! Vocab:', allVocab.length, 'Phrases:', phrases.length);
